io/plugins/eu.esdihumboldt.hale.io.xsd/src/eu/esdihumboldt/hale/io/xsd/reader/XmlSchemaReader.java

```diff
@@ -963,6 +965,59 @@ else if (object instanceof XmlSchemaParticle) {
 			}
 			// </sequence>
 		}
+		else if (particle instanceof XmlSchemaAll) {
+			// <all>
+			XmlSchemaAll all = (XmlSchemaAll) particle;
+
+			// Declaring group must be the type, <all> groups can't be
+			// children of other groups.
+			if (!(declaringGroup instanceof XmlTypeDefinition)) {
+				reporter.error(
+						"<xs:all> groups that are not immediate children of a type definition are not supported.");
```

Comment:
Is this not supported here or is that not valid in XML Schema? The message and the comment seem to hint at different things.

Reply:
It's not valid in XML Schema as `<all>` can't be put inside of `<sequence>`s or `<choice>`s ([source](https://www.w3.org/TR/xmlschema-1/#declare-contentModel)). I'll update the error message to better reflect that.